src/gui/usbipd_gui.rs

```diff
@@ -166,12 +166,20 @@ impl UsbipdGui {
         for device in devices {
             let device_name = device.description.as_deref();
             let vid_pid = device.vid_pid();
-            let description = device_name.map(|s| s.to_string()).unwrap_or(
+            let mut description: String = device_name.map(|s| s.to_string()).unwrap_or(
                 vid_pid
                     .clone()
                     .unwrap_or_else(|| "Unknown Device".to_string()),
             );
-
+            // if len > 32, truncate and add bus id
+            if description.len() > 32 { // @todo use settings replace magic number
+            description.truncate(32);
+                    let bus_id = device
+                        .bus_id
+                        .clone()
+                        .unwrap_or_else(|| "Unknown Bus".to_string());
+                    description.push_str(&format!("..(Bus:{})", bus_id));
+                }
```

Comment:
I think this is not the desired behavior. If the goal is to establish a max string length, truncating if necessary, and adding unique ID (bus ID), then the length of that metadata must be taken into account. The present implementation can result in a longer string with less information.

The truncation point is not the same as the desired max string length. The truncation point must be max_length - length(id_string).

- description is greater than max length
- format the ID string (... BUS-ID)
- truncate and concat at description[max_length - length(id_string)]

This will prevent the fix from being longer than the problem.